### ISSUE TITLE
Caching performance issue with Gitlab CI/CD

### ISSUE THREAD

**Issue by DivadNojnarg** [rstudio/renv#916]
Hi there,

In the documentation, there is a starter Gitlab template:

```yaml
variables:
  RENV_CONFIG_REPOS_OVERRIDE: "http://cran.r-project.org"
  RENV_PATHS_CACHE: ${CI_PROJECT_DIR}/cache
  RENV_PATHS_LIBRARY: ${CI_PROJECT_DIR}/renv/library

cache:
  key: ${CI_JOB_NAME}
  paths:
    - ${RENV_PATHS_CACHE}
    - ${RENV_PATHS_LIBRARY}

before_script:
  - < ... other pre-deploy steps ... >
  - Rscript -e "if (!requireNamespace('renv', quietly = TRUE)) install.packages('renv')"
  - Rscript -e "renv::restore()"
```

Most of the time, the ${RENV_PATHS_CACHE} has more than > 10000 files in my projects. This leads to very long time for job completion (> 2h if not timeout before) and is not acceptable for people in my team. 

Is there any rational behind storing this element in the cache?

I would argue that only storing the renv library is enough in most cases but maybe I missed something and would be curious to know the reason. 

If storing the renv library is enough, I'd like to add a quick note in the documentation about this possible performance issue to be aware of. it might be misleading for people beginning with the template. 

Let me know your thought.

Thanks

David

**Comment by bartekch** [comment]
@DivadNojnarg I wouldn't  answer your question, but I'm a bit surprised that it takes so long in your case. I'm using the same template, number of files in cache is just below 10k, and job takes couple of minutes, but restoring and creating cache take ~10 seconds each, so it's quite quick (run on Gitlab shared runner). Is it cache restore/create step that takes so long?

**Comment by kevinushey** [comment]
By default, the `renv` library is just a set of symlinks into the package cache, so only storing the `renv` library directory likely would not suffice. You normally need both the library and the package cache.

If you can provide more detail (e.g. the output from one of these Gitlab runs) that would be helpful.

**Comment by vladimirobucina** [comment]
@kevinushey I've done the same thing as in [documentation](https://rstudio.github.io/renv/articles/ci.html#gitlab-ci). I would expect it to run installation only once, install packages to `renv/library` and save cache for all jobs, but it seems to me like it's running installation almost every time (which execution time shows, both jobs took around 8 minutes to finish). I would send you the CI logs, but they're too big to copy paste here.

**Comment by hadley** [comment]
Closing because it's been almost a year so you've presumably resolved this problem with your CI/CD pipeline. If not, please feel free to file a new issue with more details and we'll take a look.
